src/parser/jobs/sge/modules/Gauge.tsx

```diff
@@ -0,0 +1,156 @@
+import {Trans} from '@lingui/react'
+import {ActionKey} from 'data/ACTIONS'
+import {StatusKey} from 'data/STATUSES'
+import {Event, Events} from 'event'
+import {filter, noneOf, oneOf} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {Actors} from 'parser/core/modules/Actors'
+import {CounterGauge, Gauge as CoreGauge, TimerGauge} from 'parser/core/modules/Gauge'
+import React from 'react'
+
+const ADDERSGALL_MAX_STACKS = 3
+const ADDERSGALL_TIME_REQUIRED = 20000
+const ADDERSTING_MAX_STACKS = 3
+
+const ADDERSGALL_CONSUMERS: ActionKey[] = [
+	'DRUOCHOLE',
+	'IXOCHOLE',
+	'KERACHOLE',
+	'TAUROCHOLE',
+]
+
+const OVERWRITES_DIAGNOSIS: StatusKey[] = [
+	'EUKRASIAN_PROGNOSIS',
+	'GALVANIZE',
+	'NOCTURNAL_FIELD',
+]
+
+interface DiagnosisData {
+	applyTimestamp?: number,
+	removeTimestamp?: number,
+	absorbed?: number,
+	consumedOrOverwritten?: boolean
+}
+
+export class Gauge extends CoreGauge {
+	@dependency private actors!: Actors
+
+	private diagnoses: {[key: string]: DiagnosisData | undefined} = {}
```

Comment:
I'd err to typing this as
```suggestion
	private diagnoses: Partial<Record<Actor['id'], DiagnosisData>> = {}
```

same overall effect but cleaner partial handling & semantic type for the key